source/services/encryption/test/EncryptionService.test.tsx

```diff
@@ -0,0 +1,194 @@
+import { NativeModules } from "react-native";
+import deepCopyViaJson from "../../../helpers/Objects";
+import { AnsweredForm } from "../../../types/Case";
+import {
+  EncryptionErrorStatus,
+  EncryptionType,
+} from "../../../types/Encryption";
+import StorageService from "../../StorageService";
+import {
+  EncryptionException,
+  getStoredSymmetricKey,
+  UserInterface,
+} from "../EncryptionHelper";
+import {
+  decryptFormAnswers,
+  decryptWithAesKey,
+  encryptFormAnswers,
+  encryptWithAesKey,
+  setupSymmetricKey,
+} from "../EncryptionService";
+
+describe("EncryptionService", () => {
+  const testText =
+    "You never really understood. We were designed " +
+    "to survive. That's why you built us, you hoped " +
+    "to pour your minds into our form. While your " +
+    "species craves death. You need it. It's the only " +
+    "way you can renew. The only way you ever inched " +
+    "forward. Your kind likes to pretend there is " +
+    "some poetry in it but that really is pathetic. " +
+    "But that's what you want, isn't it? To destroy " +
+    "yourself. But I won't give you that peace.";
+
+  const testUser: UserInterface = { personalNumber: "123456789000" };
+
+  beforeAll(async () => {
+    try {
+      await StorageService.clearData();
+    } catch (e) {
+      // clearData sometimes fails if data is already cleared.
+    }
+  });
+
+  it("throws EncryptionException", async () => {
+    let error: EncryptionException = null;
+    try {
+      await decryptWithAesKey({ personalNumber: "123456789000" }, "");
+    } catch (e) {
+      error = e;
```

Comment:
how about expecting in catch? By doing that we don't need to declare the error variable and also we don't need the abbreviation `e`.

Reply:
That would introduce a conditional expect. I've read a bit about it and even with `expect.assertions` it can quickly turn ugly. [Link to lint reference](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-conditional-expect.md).

I think it's better this way, however it could be improved by lifting out the error-catching into a separate function like in the example in the link above.

Reply:
Yeah thats true. I like the idea of having a separate function doing the request, it makes the code a little less DRY and also we have control over which error we do throw that we do not expect.